src/_argon2_cffi_bindings/_ffi_build.py

```diff
@@ -17,9 +17,16 @@
 elif use_sse2 == "0":
     optimized = False
 else:
-    # Optimized version requires SSE2 extensions.  They have been around since
+    # Optimized version requires SSE2 extensions. They have been around since
     # 2001 so we try to compile it on every recent-ish x86.
-    optimized = platform.machine() in ("i686", "x86", "x86_64", "AMD64")
+    cibw = os.environ.get("CIBW_BUILD", "")
+    arch_flags = os.environ.get("ARCHFLAGS", "")
+    if cibw and arch_flags:
+        # cibuildwheel currently uses cross-compiling for arm64, so we have to
+        # detect it here.
+        optimized = "-arch arm64" not in arch_flags
+    else:
+        optimized = platform.machine() in ("i686", "x86", "x86_64", "AMD64")
```

Comment:
This is not cibuildwheel specific. `ARCHFLAGS` is how you control what you are targeting using setuptools. The code for it is here: https://github.com/python/cpython/blob/v3.10.0/Lib/_osx_support.py

I'd remove the `CIBW_BUILD` check, and then I'd add a darwin check (it's macOS only to the best of my knowledge). Then I'd change the check to see if any ARCHFLAGS are set that do not match the platform.machine. That way, you'll support Conda-build and also cross compiling the other direction, such as building intel from ARM.

Reply:
My thought process here is to do only magic that I control (CIBW) and leave other cross-compilers control it via `ARGON2_CFFI_USE_SSE2`. Does that make sense?

If I could set that variable from within CIBW, I'd do it that way too…

Reply:
(My concern is that if I try to control it automatically, there's guaranteed a bazillion edge cases that it will break which I don't control)

Reply:
I think having a smart default for `ARGON2_CFFI_USE_SSE2` that detects cross compiling would be best. It could be overridden, but having do the right thing if possible is nice. Correct my text on the final suggestion above as needed, though, I was thinking of this as native/non-native, rather than x86_64 specific vs. not.